contracts/interchain-token-service/src/contract/execute/mod.rs

```diff
@@ -151,20 +150,56 @@ fn apply_to_hub(
 
     match message {
         Message::InterchainTransfer(transfer) => {
-            apply_transfer(storage, source_chain, destination_chain, &transfer)
+            apply_to_transfer(storage, source_chain, destination_chain, transfer)
                 .map(Message::InterchainTransfer)?
         }
         Message::DeployInterchainToken(deploy_token) => {
-            apply_token_deployment(storage, &source_chain, &destination_chain, deploy_token)
+            apply_to_token_deployment(storage, &source_chain, &destination_chain, deploy_token)
                 .map(Message::DeployInterchainToken)?
         }
     }
     .then(Result::Ok)
 }
 
+fn apply_to_transfer(
+    storage: &mut dyn Storage,
+    source_chain: ChainNameRaw,
+    destination_chain: ChainNameRaw,
+    transfer: InterchainTransfer,
+) -> Result<InterchainTransfer, Error> {
+    interceptors::subtract_supply_amount(storage, &source_chain, &transfer)?;
+    let transfer = interceptors::apply_scaling_factor_to_amount(
+        storage,
+        &source_chain,
+        &destination_chain,
+        transfer,
+    )?;
+    interceptors::add_supply_amount(storage, &destination_chain, &transfer)?;
+
+    Ok(transfer)
```

Comment:
here `and_then` might look better now 😆

Reply:
my biggest issue before was that it was just used to unpack the result, but the passed value was ignored. With the new fluid interface that is no longer the case

Reply:
```suggestion
    interceptors::apply_scaling_factor_to_amount(
        storage,
        &source_chain,
        &destination_chain,
        transfer,
    )
    .inspect(interceptors::add_supply_amount(storage, &destination_chain, transfer))
```